src/d/actor/d_a_fganon.cpp

```diff
@@ -5,195 +5,2884 @@
 
 #include "d/actor/d_a_fganon.h"
 #include "d/d_procname.h"
+#include "JSystem/JUtility/JUTReport.h"
+#include "d/actor/d_a_boko.h"
+#include "d/actor/d_a_mant.h"
+#include "d/actor/d_a_player.h"
+#include "d/d_bg_s_lin_chk.h"
+#include "d/d_cc_uty.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_meter.h"
+#include "d/d_s_play.h"
+#include "d/d_snap.h"
+#include "d/res/res_fganon.h"
+#include "d/res/res_pgsw.h"
+#include "f_op/f_op_camera.h"
+
+#include "weak_bss_936_to_1036.h" // IWYU pragma: keep
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+fganon_class* master;
+u8 hio_set;
+daFganon_HIO_c l_HIO;
 
 /* 000000EC-000001A0       .text __ct__14daFganon_HIO_cFv */
 daFganon_HIO_c::daFganon_HIO_c() {
-    /* Nonmatching */
+    mNo = -0x1;
+    m05 = 0;
+    m06 = 0;
+    m08 = 60;
+    m0A = 500;
+    m0C = 1500.0f;
+    m10 = 400.0f;
+    m14 = 1500.0f;
+    m18 = 400.0f;
+    m1C = 45.0f;
+    m20 = 45.0f;
+    m24 = 2.0f;
+    m28 = 2.0f;
+    m2C = 30.0f;
+    m30 = -270.0f;
+    m34 = -600.0f;
+    m38 = 270.0f;
+    m3C = 50;
+    m3E = 1;
+    m40 = 32;
+    m44 = 600.0f;
+    m48 = 150;
 }
 
 /* 000001A0-000002C8       .text anm_init__FP12fganon_classifUcfi */
-void anm_init(fganon_class*, int, float, unsigned char, float, int) {
-    /* Nonmatching */
+void anm_init(fganon_class* i_this, int bckFileIdx, f32 morf, u8 loopMode, f32 speed, int soundFileIdx) {    
+    if (soundFileIdx >= 0) {
+        void* soundAnm = dComIfG_getObjectRes("Fganon", soundFileIdx);
+        J3DAnmTransform* bckAnm = (J3DAnmTransform*)dComIfG_getObjectRes("Fganon", bckFileIdx);
+        i_this->mpMorf->setAnm(bckAnm, loopMode, morf, speed, 0.0f, -1.0f, soundAnm);
+    }
+    else {
+        J3DAnmTransform* bckAnm = (J3DAnmTransform*)dComIfG_getObjectRes("Fganon", bckFileIdx);
+        i_this->mpMorf->setAnm(bckAnm, loopMode, morf, speed, 0.0f, -1.0f, NULL);
+    }
+    return;
 }
 
 /* 000002C8-00000338       .text tama_draw__FP12fganon_class */
-void tama_draw(fganon_class*) {
-    /* Nonmatching */
+/* Renders PG's Energy Ball model */
+void tama_draw(fganon_class* i_this) {
+    if (i_this->m671 != 0) {
+        J3DModel* pModel = i_this->EnergySphereModel;
+        i_this->mpBrkAnm3->entry(pModel->getModelData());
+        i_this->mpBtkAnm->entry(pModel->getModelData());
+        mDoExt_modelUpdateDL(pModel);
+    }
+    return;
 }
 
 /* 00000338-00000420       .text daFganon_Draw__FP12fganon_class */
-static BOOL daFganon_Draw(fganon_class*) {
-    /* Nonmatching */
+/* Renders PG's NPC model */
```

Comment:
This comment still calls his sword model his "NPC model". Also, this function renders all parts of PG, not just his sword (mpMorf is his actual body, mpKenModel is his sword, then it calls tama_draw which renders the ball)